app/service/SchemeServiceImpl.scala

```diff
@@ -25,26 +25,47 @@ import play.api.Logger
 import play.api.libs.json.{JsValue, Json}
 import play.api.mvc.RequestHeader
 import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier, HttpException}
+import utils.validationUtils._
+
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
-import utils.validationUtils._
 
 class SchemeServiceImpl @Inject()(desConnector: DesConnector,
                                   auditService: AuditService, appConfig: AppConfig) extends SchemeService with SchemeAuditService {
 
-  override def registerPSA(json: JsValue)
-                          (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, rh: RequestHeader): Future[Either[HttpException, JsValue]] = {
+  override def registerPSA(json: JsValue)(
+    implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, rh: RequestHeader): Future[Either[HttpException, JsValue]] = {
+
+    getPensionSchemeAdministrator(json) { pensionSchemeAdministrator =>
+      val psaJsValue = Json.toJson(pensionSchemeAdministrator)(PensionSchemeAdministrator.psaSubmissionWrites)
+      Logger.debug(s"[PSA-Registration-Outgoing-Payload]$psaJsValue")
+      desConnector.registerPSA(psaJsValue) andThen sendPSASubscriptionEvent(pensionSchemeAdministrator, psaJsValue)(auditService.sendEvent)
+    }
+
+  }
+
+  override def updatePSA(psaId: String, json: JsValue)(
+    implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader): Future[Either[HttpException, JsValue]] = {
+
+    getPensionSchemeAdministrator(json) { pensionSchemeAdministrator =>
+      val psaJsValue = Json.toJson(pensionSchemeAdministrator)
+      Logger.debug(s"[PSA-Variation-Outgoing-Payload]$psaJsValue")
+      desConnector.updatePSA(psaId, psaJsValue)
+    }
+  }
+
+  def getPensionSchemeAdministrator(json: JsValue)(
```

Comment:
Can we call this something like convert and make return value Unit explicit?

Reply:
its not only doing conversion but also handling Try